### PR TITLE
refactor: Show template versions as timeline

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
- Extract `Timeline` component
- Refactor templateversion to return `created_by` instead of `created_by_something`

Before:
Screen Shot 2022-10-28 at 15 44 25

After (Sorry, I only have the one in storybook to use as example)
Screen Shot 2022-10-28 at 15 43 25

### CODE REVIEW

File: codersdk/templateversions.go [hunk 1]
```diff
@@ -21,8 +21,7 @@ type TemplateVersion struct {
 	Name           string         `json:"name"`
 	Job            ProvisionerJob `json:"job"`
 	Readme         string         `json:"readme"`
-	CreatedByID    uuid.UUID      `json:"created_by_id"`
-	CreatedByName  string         `json:"created_by_name"`
+	CreatedBy      User           `json:"created_by"`
```
Comment (kylecarbs) [inline codersdk/templateversions.go:24]: I'd make this `Creator` instead

**Review by kylecarbs — APPROVED** [review]

File: codersdk/templateversions.go (hunk 1 above)
Comment (BrunoQuaresma) [reply codersdk/templateversions.go:24]: Do you have a strong opinion about that? I like the `CreatedBy` tho.

File: coderd/database/migrations/000066_template_version_created_by.up.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+ALTER TABLE template_versions ALTER COLUMN created_by SET NOT NULL
```
Comment (coadler) [inline coderd/database/migrations/000066_template_version_created_by.up.sql:1]: This needs to also set all `NULL` values to a zero uuid.

Comment (BrunoQuaresma) [reply coderd/database/migrations/000066_template_version_created_by.up.sql:1]: How this can be made?

Comment (coadler) [reply coderd/database/migrations/000066_template_version_created_by.up.sql:1]: set created by to `'00000000-0000-0000-0000-000000000000'::uuid` when the column is null

Comment (BrunoQuaresma) [reply coderd/database/migrations/000066_template_version_created_by.up.sql]: Where and how should I do that? I'm not understanding 😕

Comment (coadler) [reply coderd/database/migrations/000066_template_version_created_by.up.sql]: before the alter table
`UPDATE template_versions SET created_by = '00000000-0000-0000-0000-000000000000'::uuid WHERE created_by IS NULL`

Comment (BrunoQuaresma) [reply coderd/database/migrations/000066_template_version_created_by.up.sql]: Ahhh I got it. Thanks for the help!

Comment (coadler) [reply coderd/database/migrations/000066_template_version_created_by.up.sql]: also needs a `BEGIN` and `END` to make it a transaction

Comment (BrunoQuaresma) [reply coderd/database/migrations/000066_template_version_created_by.up.sql]: `BEGIN` and `COMMIT` right?

Comment (coadler) [reply coderd/database/migrations/000066_template_version_created_by.up.sql]: ah yea you're right!

**Comment by BrunoQuaresma** [comment]
I would love some help/guidance on how to fix the test. I don't know what is causing the error below:
```
Error Trace:	/home/runner/work/coder/coder/coderd/templateversions_test.go:494
        	Error:      	Received unexpected error:
        	            	do:
        	            	    github.com/coder/coder/codersdk.(*Client).Request
        	            	        /home/runner/work/coder/coder/codersdk/client.go:100
        	            	  - Get "http://localhost:34269/api/v2/templates/373ac77a-20e4-4d38-9c47-e8df45e295b6/versions": context deadline exceeded
        	Test:       	TestTemplateVersionsByTemplate/Get

```

File: coderd/templateversions.go [hunk 3]
```diff
@@ -523,15 +522,15 @@ func (api *API) templateVersionsByTemplate(rw http.ResponseWriter, r *http.Reque
 				})
 				return err
 			}
-			createdByName, err := getUsernameByUserID(ctx, store, version.CreatedBy)
+			user, err := api.Database.GetUserByID(ctx, version.CreatedBy)
```
Comment (sreya) [inline coderd/templateversions.go:525]: @BrunoQuaresma this is the line you need to change. Make this `store.GetUserByID`

Comment (sreya) [reply coderd/templateversions.go:525]: I would also remove this transaction or consult with whoever wrote it to ensure that it is necesssary

Comment (BrunoQuaresma) [reply coderd/templateversions.go]: Fixed! 👍

**Pull request merged**